src/node.cpp

```diff
@@ -131,11 +131,14 @@ Status Node::setInputs(const Node& dependency, TensorMap& inputs, NodeSessionMet
             dependency.getName(),
             current_node_input_name,
             dependency_output_name);
-        auto status = nodeSession->setInput(current_node_input_name, it->second, shardId);
+        auto status = nodeSession->setInput(current_node_input_name, it->second.getActualTensor(), shardId);
```

Comment:
Wouldn't it make sense to instead of asking of original tensor for source just pass this wrapper into InputHandler?

Reply:
If we saved the wrapper in InputHandler we would need to pay for creating empty ov::Tensor() (source tensor) even if there is no demultiplexer in pipeline

Reply:
Adjusted after sync.